src/main/java/InputHandler.java

```diff
@@ -0,0 +1,127 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * Handles input in main in Duke.java. Receives String from Scanner in main
+ * Processes the input into 7 categories: Todo, Event, Deadline, list, mark, unmark, bye and throws error
+ */
+public class InputHandler {
+    private ArrayList<Task> arr;
+    public InputHandler() {
+        this.arr = new ArrayList<>();
+    }
+
+
+    /**
+     *
+     * @param input String input from main in Duke.java, from user input
+     * @return boolean representing isChatEnded variable in main. If "bye" command is given, a true boolean is returned, else false is returned
+     * @throws DukeException Invalid input types, or unrecognisable commands
+     */
+    public boolean handleInput(String input) throws DukeException {
+        String[] splitInput = input.split(" ");
+        String inputCommand = splitInput[0];
+        switch (inputCommand) {
+            case "todo":
```

Comment:
I didn't like the indentation after the switch, are you sure this complies with the coding standard? Consider leaving no indentation for the case statements.